src/gpu/mod.rs

```diff
@@ -37,3 +37,26 @@ use ocl::Device;
 lazy_static::lazy_static! {
     pub static ref GPU_NVIDIA_DEVICES: Vec<Device> = get_devices(GPU_NVIDIA_PLATFORM_NAME).unwrap_or_default();
 }
+
+#[derive(Debug)]
+pub struct LockedFile(File);
+
+use std::fs::File;
+
+use std::io;
+pub const LOCK_NAME: &str = "/tmp/bellman.lock";
```

Comment:
@dignifiedquire If the host goes down ungracefully (power outage) after it has obtained the lock, does this lock leak?

Reply:
Let me ask differently: Do you think it’s safe to rely on the OS wiping /tmp on reboot?

Reply:
@laser There are two different structures in this PR, one is a file-lock (The one you mentioned) used to prevent two different processes to run the `create_proof` function at the same time. The existence of this file-lock (`/tmp/bellman.lock`) does not specify if the lock is acquired or released, but is a mutex-like structure which its state is managed by the operating-system itself (The file-locks associated to a process are automatically released when the process gets killed, gracefully or ungracefully) so you can be sure this part of the code is guaranteed to be safe.

Reply:
@laser As for the second structure (The one used to signal other processes to kill their kernels) yes, a crash would break things a bit, but I think its reasonable to assume the process with more priority has to manage the situation in case something like that happens.